src/app/utils/field-validation/validators/numberValidator.ts

```diff
@@ -59,29 +62,64 @@ const maxLengthValidator: NumberValidatorConstructor =
 const exactLengthValidator: NumberValidatorConstructor =
   (numberField) => (response) => {
     const { answer } = response
-    const { customVal } = numberField.ValidationOptions
+    const { customVal } = numberField.ValidationOptions.LengthValidationOptions
     return !customVal || answer.length === customVal
       ? right(response)
       : left(`NumberValidator:\t answer does not match custom exact length`)
   }
 
 /**
- * Returns the appropriate validation function
- * based on the number validation option selected.
+ * Returns the appropriate number length validation function
+ * based on the number length validation option selected.
  */
 const getNumberLengthValidator: NumberValidatorConstructor = (numberField) => {
-  switch (numberField.ValidationOptions.selectedValidation) {
-    case NumberSelectedValidation.Min:
+  switch (
+    numberField.ValidationOptions.LengthValidationOptions
+      .selectedLengthValidation
+  ) {
+    case NumberSelectedLengthValidation.Min:
       return minLengthValidator(numberField)
-    case NumberSelectedValidation.Max:
+    case NumberSelectedLengthValidation.Max:
       return maxLengthValidator(numberField)
-    case NumberSelectedValidation.Exact:
+    case NumberSelectedLengthValidation.Exact:
       return exactLengthValidator(numberField)
     default:
       return right
   }
 }
 
+/**
+ * Returns a validation function to check if number is
+ * within the number range specified.
+ */
+const getNumberRangeValidator: NumberValidatorConstructor =
+  (numberField) => (response) => {
+    const val = Number(response.answer)
+    const { customMin, customMax } =
+      numberField.ValidationOptions.RangeValidationOptions
+    const isWithinMinimum = !!customMin && customMin <= val
+    const isWithinMaximum = !!customMax && val <= customMax
```

Comment:
Same comments here as above in the frontend field validator function, re: NaN check, Boolean cast.

Reply:
Related to above comments

Reply:
I have made the changes regarding the Boolean cast!

Regarding the NaN check, I just realized that we already validate that `response.answer` is not `''` and a non-negative integer from the upstream validators. Since the composition of these validators are explicit, would the NaN check be necessary? I think we should make use of the assertions about `response.answer` from the previous validators, and add in a comment such as `// Chained validators ensure that the casting is valid`

Of course, the checks are not complex/expensive so we could always add them in. Let me know what are your thoughts on this!

Reply:
I think that's a fair argument. I'm ok with either way. Thanks for addressing the issue!